cwms-data-api/src/main/java/cwms/cda/api/LookupTypeController.java

```diff
@@ -0,0 +1,217 @@
+                    },
+                    required = true),
+            description = "Update CWMS Lookup Type",
+            method = HttpMethod.PATCH,
+            tags = {TAG},
+            responses = {
+                    @OpenApiResponse(status = STATUS_204, description = "Lookup Type successfully stored to CWMS.")
+            }
+    )
+    @Override
+    public void update(Context ctx, String name) {
+        String category = requiredParam(ctx, CATEGORY);
+        String prefix = requiredParam(ctx, PREFIX);
+        try (Timer.Context ignored = markAndTime(UPDATE)) {
+            String acceptHeader = ctx.req.getContentType();
+            String formatHeader = acceptHeader != null ? acceptHeader : Formats.JSON;
+            ContentType contentType = Formats.parseHeader(formatHeader);
+            LookupType lookupType = Formats.parseContent(contentType, ctx.body(), LookupType.class);
+            lookupType.validate();
+            DSLContext dsl = getDslContext(ctx);
+            LookupTypeDao dao = new LookupTypeDao(dsl);
+            dao.updateLookupType(category, prefix, lookupType);
+            ctx.status(HttpServletResponse.SC_OK).json("Updated Lookup Type");
+        }
+    }
+
+    @OpenApi(
+            queryParams = {
+                    @OpenApiParam(name = CATEGORY, description = "Specifies the category id of the lookup type to be deleted."),
+                    @OpenApiParam(name = PREFIX, description = "Specifies the prefix of the lookup type to be deleted."),
+                    @OpenApiParam(name = OFFICE, description = "Specifies the owning office of the lookup type to be deleted."),
+            },
+            description = "Delete CWMS Lookup Type",
+            method = HttpMethod.DELETE,
+            tags = {TAG},
+            responses = {
+                    @OpenApiResponse(status = STATUS_204, description = "Lookup Type successfully deleted from CWMS."),
+                    @OpenApiResponse(status = STATUS_404, description = "Based on the combination of inputs provided the lookup type was not found.")
+            }
+    )
+    @Override
+    public void delete(Context ctx, @NotNull String displayValue) {
+        String officeId = requiredParam(ctx, OFFICE);
+        String category = requiredParam(ctx, CATEGORY);
+        String prefix = requiredParam(ctx, PREFIX);
+        try (Timer.Context ignored = markAndTime(DELETE)) {
+            DSLContext dsl = getDslContext(ctx);
+            LookupTypeDao dao = new LookupTypeDao(dsl);
+            List<LookupType> lookupTypes = dao.retrieveLookupTypes(category, prefix, officeId);
+            LookupType lookupType = lookupTypes.stream()
+                    .filter(lt -> lt.getDisplayValue().equals(displayValue))
+                    .findFirst()
+                    .orElse(null);
+            if (lookupType != null) {
+                dao.deleteLookupType(category, prefix, lookupType);
```

Comment:
Does this call not fail in a manageable way and that's why you're doing the lookup first?

Reply:
Refactored this lookup into the DAO